### PR TITLE
WT-3772 Do not increment missed pre-allocated logs counter when a hot backup …

### PR DESCRIPTION
**Pull request by turchanov**
…is in progress since we deliberately not using pre-allocated log files in that case.

To resolve https://jira.mongodb.org/browse/WT-3772

### CODE REVIEW

**Comment by wtbuild** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by agorrod** [comment]
autobuild: test this please

**Comment by turchanov** [comment]
It seems there is a problem with Jenkins as I have built the sources without any problems.

**Comment by agorrod** [comment]
The failure is from a style-checking script. Could you run `cd dist && sh s_all` and fix/push any changes?

**Comment by turchanov** [comment]
There was formatting error +  I slightly modified logic of a patch.
Do you want me to squash commits into a single commit or multiple commits will be fine?

**Comment by agorrod** [comment]
Multiple commits are fine

**Comment by turchanov** [comment]
I pushed commits to this PR but autobuild didn't start. Do I have to create a new pull request? I'm a lillte bit confused.

**Comment by agorrod** [comment]
autobuild: test this please

**Comment by agorrod** [comment]
There is a protection in place to avoid unknown users attempting to break our automated testing setup by sending in malicious pull requests. I've triggered the tests.

**Comment by turchanov** [comment]
Build failed again and it is somewhere way off my changes. I only modified "src/log/log.c" and it compiles just fine.

**Comment by michaelcahill** [comment]
autobuild: retest this please

**Comment by michaelcahill** [comment]
@turchanov, our standard practice is to open pull requests against the `develop` branch, which is what the automated tests expect.  I've changed the base for your pull request, so the new round of tests should have a better chance of completing.

**Comment by michaelcahill** [comment]
autobuild: retest this please

File: src/log/log.c [hunk 1]
```diff
@@ -976,7 +976,14 @@ __log_newfile(WT_SESSION_IMPL *session, bool conn_open, bool *created)
...
 	 */
 	if (create_log) {
-		log->prep_missed++;
+		/*
+		 * Do not increment missed prep logs counter when a hot backup
+		 * is in progress and pre-allocation is enabled since
+		 * we deliberately not using prep log files in that case
+		 * (see comments above).
+		 */
+		if ((conn->log_prealloc == 0) || !conn->hot_backup)
```
Comment (sueloverso) [inline src/log/log.c]: Thank you @turchanov. I think the original fix is clearer (i.e. only check !conn->hot_backup in the conditional). Also, can you slightly modify the comment, as we generally use full, not abbreviated words and sentences in the comments. I suggest something like this:
"Only increment the missed pre-allocated file counter if a hot backup is not in progress. We are deliberately not using pre-allocated log files during backup (see comment above)."

**Comment by agorrod** [comment]
Thanks again for your contribution @turchanov - and congratulations on navigating our automated testing minefield!

**Pull request merged**